clients/include/device_batch_vector.hpp

```diff
@@ -0,0 +1,236 @@
+    {
+        return this->m_size_vector;
+    };
+
+    //!
+    //! @brief Returns the value of batch_count.
+    //!
+    inline rocblas_int batch_count() const noexcept
+    {
+        return this->m_batch_count;
+    };
+
+    //!
+    //! @brief Access to device data.
+    //! @return Pointer to the device data.
+    //!
+    inline T** ptr_on_device() noexcept
+    {
+        return this->m_device_data;
+    }
+
+    //!
+    //! @brief Const access to device data.
+    //! @return Const pointer to the device data.
+    //!
+    inline const T* const* ptr_on_device() const noexcept
+    {
+        return this->m_device_data;
+    }
+
+    //!
+    //! @brief Random access.
+    //! @param batch_index The batch index.
+    //! @return Pointer to the array on device.
+    //!
+    inline T* operator[](rocblas_int batch_index) noexcept
+    {
+        assert(0 <= batch_index && this->m_batch_count > batch_index);
+
+        return this->m_data[batch_index];
+    }
+
+    //!
+    //! @brief Constant random access.
+    //! @param batch_index The batch index.
+    //! @return Constant pointer to the array on device.
+    //!
+    inline const T* operator[](rocblas_int batch_index) const noexcept
+    {
+        assert(0 <= batch_index && this->m_batch_count > batch_index);
+
+        return this->m_data[batch_index];
+    }
+
+    //!
+    //! @brief Const cast of the data on host.
+    //!
+    inline operator const T* const*() const noexcept
+    {
+        return this->m_data;
+    };
+
+    //!
+    //! @brief Cast of the data on host.
+    //!
+    inline operator T**() noexcept
+    {
+        return this->m_data;
+    }
+
+    //!
+    //! @brief Tell whether ressources allocation failed.
+    //!
+    inline explicit operator bool() const noexcept
+    {
+        return this->m_data != nullptr;
+    }
+
+    //!
+    //! @brief Copy from a host batched vector.
+    //! @param that The host_batch_vector to copy.
+    //!
+    inline hipError_t transfer_from(const host_batch_vector<T>& that) noexcept
+    {
+        //
+        // Check sizes.
+        //
+        if((this->batch_count() != that.batch_count()) || (this->size() != that.size()))
+        {
+            return hipErrorInvalidContext;
+        }
+
+        //
+        // Copy each vector.
+        //
+        for(rocblas_int batch_index = 0; batch_index < this->m_batch_count; ++batch_index)
+        {
+            auto hip_err = hipMemcpy((*this)[batch_index],
+                                     that[batch_index],
+                                     sizeof(T) * this->m_size_vector,
+                                     hipMemcpyHostToDevice);
+            if(hipSuccess != hip_err)
+            {
+                return hip_err;
+            }
+        }
+
+        return hipSuccess;
+    }
+
+private:
+    rocblas_int m_batch_count{0};
+    size_t      m_size_vector{0};
+    T**         m_data{nullptr};
+    T**         m_device_data{nullptr};
+
```

Comment:
`{}` works for all 4 too.
```
    rocblas_int m_batch_count{};
    size_t      m_size_vector{};
    T**         m_data{};
    T**         m_device_data{};
```
minimalistic, but maybe not as documenting.

Reply:
I prefer explicit values rather than implicit ones.

Reply:
`{}` is pretty explicit. Leaving `{}` out is implicit.

Reply:
Let's use {} then.